cranelift/reader/src/parser.rs

```diff
@@ -2565,14 +2538,14 @@ impl<'a> Parser<'a> {
                     Ok(RunCommand::Run(invocation, comparison, expected))
                 } else if sig.params.is_empty()
                     && sig.returns.len() == 1
-                    && sig.returns[0].value_type.is_bool()
+                    && sig.returns[0].value_type.is_int()
                 {
                     // To match the existing run behavior that does not require an explicit
-                    // invocation, we create an invocation from a function like `() -> b*` and
-                    // compare it to `true`.
+                    // invocation, we create an invocation from a function like `() -> i*` and
+                    // compare it to not `false`.
```

Comment:
s/false/0/ here (or "and require the return value to be nonzero")?

Reply:
Great point. I like the latter phrasing, and will make that change.